Error: The non-abstract class 'ChipsInputState' is missing implementations for these members:  - TextInputClient.performSelector

## Environment

**Package version:**  

<details>
  <summary>Flutter doctor</summary>

<!-- Paste the output of running `flutter doctor -v` below inside ``` and ``` -->
```
C:\src\flutter\bin\flutter.bat doctor --verbose

[√] Flutter (Channel master, 3.1.0-0.0.pre.2166, on Microsoft Windows
[Version 10.0.14393], locale en-IN)

• Flutter version 3.1.0-0.0.pre.2166 on channel master at C:\src\flutter

• Upstream repository https://github.com/flutter/flutter.git

• Framework revision 87f1684476 (30 hours ago), 2022-08-08 23:00:11 -0700

• Engine revision b6dd604d90

• Dart version 2.19.0 (build 2.19.0-76.0.dev)

• DevTools version 2.16.0


[√] Android toolchain - develop for Android devices (Android SDK version
32.1.0-rc1)

• Android SDK at C:\Users\Snow\AppData\Local\Android\sdk

• Platform android-32, build-tools 32.1.0-rc1

• Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java

• Java version OpenJDK Runtime Environment (build 11.0.11+9-b60-7590822)

• All Android licenses accepted.


[√] Chrome - develop for the web

• Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe


[√] Visual Studio - develop for Windows (Visual Studio Community 2019
16.11.14)

• Visual Studio at C:\Program Files (x86)\Microsoft Visual
Studio\2019\Community

• Visual Studio Community 2019 version 16.11.32428.217

• Windows 10 SDK version 10.0.16299.0


[√] Android Studio (version 2021.1)

• Android Studio at C:\Program Files\Android\Android Studio

• Flutter plugin can be installed from:

https://plugins.jetbrains.com/plugin/9212-flutter

• Dart plugin can be installed from:

https://plugins.jetbrains.com/plugin/6351-dart

• Java version OpenJDK Runtime Environment (build 11.0.11+9-b60-7590822)


[√] VS Code (version 1.70.0)

• VS Code at C:\Users\Snow\AppData\Local\Programs\Microsoft VS Code

• Flutter extension version 3.44.0


[√] Connected device (2 available)

• Windows (desktop) • windows • windows-x64 • Microsoft Windows [Version
10.0.14393]

• Chrome (web) • chrome • web-javascript • Google Chrome 104.0.5112.81


[√] HTTP Host Availability

• All required HTTP hosts are available


• No issues found!

Process finished with exit code 0
```

</details>

<details>
  <summary>Code sample</summary>

<!--
      Please create a minimal reproducible sample that shows the problem
      and attach it below between the lines with the backticks.

      You can use a example code or Flutter sample app

      Without this we will unlikely be able to progress on the issue, and because of that
      we regretfully will have to close it.
-->

```dart
import 'package:flutter/material.dart';
import 'package:form_builder_extra_fields/form_builder_extra_fields.dart';

void main() {
  runApp(const MyApp());
  }
  
  class MyApp extends StatelessWidget {
  const MyApp({super.key});
  
  // This widget is the root of your application.
  [@OverRide](https://github.com/OverRide)
  Widget build(BuildContext context) {
    return MaterialApp(
    title: 'Flutter Demo',
      home: FormBuilderTouchSpin(
      name: 'Volume',
      ),
    );
  }
}
```

</details>

## Description

**Expected behavior:**

**Current behavior:**

## Steps to reproduce

1. This
2. Than that
3. Then

## Images <!-- if available, else delete -->  

## Stacktrace/Logcat <!-- if available, else delete -->


Launching lib\main.dart on moto g31 in debug mode...

Running Gradle task 'assembleDebug'...

../../AppData/Local/Pub/Cache/hosted/
pub.dartlang.org/flutter_chips_input-2.0.0/lib/src/chips_input.dart:87:7:
Error: The non-abstract class 'ChipsInputState' is missing implementations
for these members:

- TextInputClient.performSelector

Try to either

- provide an implementation,

- inherit an implementation from a superclass or mixin,

- mark the class as abstract, or

- provide a 'noSuchMethod' implementation.


class ChipsInputState<T> extends State<ChipsInput<T>>

^^^^^^^^^^^^^^^

/C:/src/flutter/packages/flutter/lib/src/services/text_input.dart:1172:8:
Context: 'TextInputClient.performSelector' is defined here.

void performSelector(String selectorName) {}

^^^^^^^^^^^^^^^

FAILURE: Build failed with an exception.

* Where:

Script 'C:\src\flutter\packages\flutter_tools\gradle\flutter.gradle' line:
1159

* What went wrong:

Execution failed for task ':app:compileFlutterBuildDebug'.
 Process 'command 'C:\src\flutter\bin\flutter.bat'' finished with non-zero
exit value 1

* Try:
 Run with --stacktrace option to get the stack trace.
 Run with --info or --debug option to get more log output.
 Run with --scan to get full insights.
* Get more help at https://help.gradle.org/

BUILD FAILED in 8s

I attached the document with the previous email, but I am putting the
details directly in this mail

Environment

OS: Widows 10

IDE : Android studio bumblebee 2021.1.1 patch 1

Package version:

Flutter doctor

C:\src\flutter\bin\flutter.bat doctor --verbose

[√] Flutter (Channel master, 3.1.0-0.0.pre.2166, on Microsoft Windows
[Version 10.0.14393], locale en-IN)

• Flutter version 3.1.0-0.0.pre.2166 on channel master at C:\src\flutter

• Upstream repository https://github.com/flutter/flutter.git

• Framework revision 87f1684476 (30 hours ago), 2022-08-08 23:00:11 -0700

• Engine revision b6dd604d90

• Dart version 2.19.0 (build 2.19.0-76.0.dev)

• DevTools version 2.16.0


[√] Android toolchain - develop for Android devices (Android SDK version
32.1.0-rc1)

• Android SDK at C:\Users\Snow\AppData\Local\Android\sdk

• Platform android-32, build-tools 32.1.0-rc1

• Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java

• Java version OpenJDK Runtime Environment (build 11.0.11+9-b60-7590822)

• All Android licenses accepted.


[√] Chrome - develop for the web

• Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe


[√] Visual Studio - develop for Windows (Visual Studio Community 2019
16.11.14)

• Visual Studio at C:\Program Files (x86)\Microsoft Visual
Studio\2019\Community

• Visual Studio Community 2019 version 16.11.32428.217

• Windows 10 SDK version 10.0.16299.0


[√] Android Studio (version 2021.1)

• Android Studio at C:\Program Files\Android\Android Studio

• Flutter plugin can be installed from:

https://plugins.jetbrains.com/plugin/9212-flutter

• Dart plugin can be installed from:

https://plugins.jetbrains.com/plugin/6351-dart

• Java version OpenJDK Runtime Environment (build 11.0.11+9-b60-7590822)


[√] VS Code (version 1.70.0)

• VS Code at C:\Users\Snow\AppData\Local\Programs\Microsoft VS Code

• Flutter extension version 3.44.0


[√] Connected device (2 available)

• Windows (desktop) • windows • windows-x64 • Microsoft Windows [Version
10.0.14393]

• Chrome (web) • chrome • web-javascript • Google Chrome 104.0.5112.81


[√] HTTP Host Availability

• All required HTTP hosts are available


• No issues found!

Process finished with exit code 0





Code

*import **'package:flutter/material.dart'*;
*import *
*'package:form_builder_extra_fields/form_builder_extra_fields.dart'*;

*void *main() {
runApp(*const *MyApp());
}

*class *MyApp *extends *StatelessWidget {
*const *MyApp({*super*.key});

*// This widget is the root of your application.*
@override
Widget build(BuildContext context) {
*return *MaterialApp(
title: *'Flutter Demo'*,
home: FormBuilderTouchSpin(
name: *'Volume'*,
),
);
}
}



Console output

Launching lib\main.dart on moto g31 in debug mode...

Running Gradle task 'assembleDebug'...

../../AppData/Local/Pub/Cache/hosted/
pub.dartlang.org/flutter_chips_input-2.0.0/lib/src/chips_input.dart:87:7:
Error: The non-abstract class 'ChipsInputState' is missing implementations
for these members:

- TextInputClient.performSelector

Try to either

- provide an implementation,

- inherit an implementation from a superclass or mixin,

- mark the class as abstract, or

- provide a 'noSuchMethod' implementation.


class ChipsInputState<T> extends State<ChipsInput<T>>

^^^^^^^^^^^^^^^

/C:/src/flutter/packages/flutter/lib/src/services/text_input.dart:1172:8:
Context: 'TextInputClient.performSelector' is defined here.

void performSelector(String selectorName) {}

^^^^^^^^^^^^^^^

FAILURE: Build failed with an exception.

* Where:

Script 'C:\src\flutter\packages\flutter_tools\gradle\flutter.gradle' line:
1159

* What went wrong:

Execution failed for task ':app:compileFlutterBuildDebug'.

> Process 'command 'C:\src\flutter\bin\flutter.bat'' finished with non-zero
exit value 1

* Try:

> Run with --stacktrace option to get the stack trace.

> Run with --info or --debug option to get more log output.

> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 8s



Please let me know if you need more details

On Wed, Aug 10, 2022 at 6:16 PM Matias de Andrea ***@***.***>
wrote:

> Where?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/flutter-form-builder-ecosystem/form_builder_extra_fields/issues/26#issuecomment-1210625540>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AGFRU5OVQKEWMCYVFN4EEDLVYOQD7ANCNFSM56AQIJRA>
> .
> You are receiving this because you were mentioned.Message ID:
> <flutter-form-builder-ecosystem/form_builder_extra_fields/issues/26/1210625540
> @github.com>
>

--- a/lib/src/widgets/flutter_chips_input/chips_input.dart
+++ b/lib/src/widgets/flutter_chips_input/chips_input.dart
@@ -288,7 +288,7 @@ class ChipsInputState<T> extends State<ChipsInput<T>>
     Future.delayed(const Duration(milliseconds: 300), () {
       WidgetsBinding.instance.addPostFrameCallback((_) async {
         final renderBox = context.findRenderObject() as RenderBox;
-        await Scrollable.of(context)?.position.ensureVisible(renderBox);
+        await Scrollable.of(context).position.ensureVisible(renderBox);
       });
     });
   }
@@ -490,4 +490,11 @@ class ChipsInputState<T> extends State<ChipsInput<T>>
 
   @override
   void removeTextPlaceholder() {}
+
+  @override
+  void didChangeInputControl(
+      TextInputControl? oldControl, TextInputControl? newControl) {}
+
+  @override
+  void performSelector(String selectorName) {}
 }

--- a/lib/src/widgets/flutter_chips_input/suggestions_box_controller.dart
+++ b/lib/src/widgets/flutter_chips_input/suggestions_box_controller.dart
@@ -14,7 +14,7 @@ class SuggestionsBoxController {
   void open() {
     if (_isOpened) return;
     assert(overlayEntry != null);
-    Overlay.of(context)?.insert(overlayEntry!);
+    Overlay.of(context).insert(overlayEntry!);
     _isOpened = true;
   }
 

